src/main/scala/io/qbeast/spark/QbeastTable.scala

```diff
@@ -43,17 +43,35 @@ class QbeastTable private (
     extends Serializable
     with StagingUtils {
 
+  /**
+   * The Delta Log for the Table
+   */
   private val deltaLog: DeltaLog = DeltaLog.forTable(sparkSession, tableID.id)
 
+  /**
+   * The QbeastSnapshot for the Table
+   * @return
+   */
   private def qbeastSnapshot: DeltaQbeastSnapshot = {
     val snapshot = deltaLog.update()
     DeltaQbeastSnapshot(snapshot)
   }
 
+  /**
+   * The IndexedTable representation of the Table
+   * @return
+   */
   private def indexedTable: IndexedTable = indexedTableFactory.getIndexedTable(tableID)
 
+  /**
+   * Checks if the revision is available in the table.
+   *
+   * If the revision is not available, or is not the staging revision, an exception is thrown.
+   * @param revisionID
+   *   the revision to check
+   */
   private def checkRevisionAvailable(revisionID: RevisionID): Unit = {
-    if (!qbeastSnapshot.existsRevision(revisionID)) {
+    if (revisionID != stagingID && !qbeastSnapshot.existsRevision(revisionID)) {
```

Comment:
Use the StagingUtil strait to asses if a revisionID is the staging revision.

Reply:
mmm, the `revisionID` value is available when extending the interface also. 

Should we keep it private?

Reply:
i'm ok with using the `isStaging` method, though

Reply:
Deleting this part